tools/container-build.sh

```diff
@@ -7,36 +7,53 @@
 
 set -ex
 
-# Without this, `go install` produces:
-# # runtime/cgo
-# gcc: error: unrecognized command-line option '-m64'
-if [ "$(uname -m)" = "arm64" ]; then
-  export DOCKER_DEFAULT_PLATFORM=linux/amd64
-fi
-
 if [ -z "${GO_VERSION}" ] ; then
   echo "GO_VERSION not set"
   exit 1
 fi
 
-ARCH="$(uname -m)"
 COMMIT_ID="$(git rev-parse --short=8 HEAD)"
 VERSION="${GO_VERSION}.$(date +%s)"
 
+# Detect architecture to build for (allow override via DOCKER_DEFAULT_PLATFORM)
+if [ -n "${DOCKER_DEFAULT_PLATFORM}" ]; then
+    PLATFORM="${DOCKER_DEFAULT_PLATFORM}"
+else
+    case "$(uname -m)" in
+        "x86_64") PLATFORM="linux/amd64" ;;
+        "aarch64"|"arm64") PLATFORM="linux/arm64" ;;
+        *) echo "Unsupported architecture: $(uname -m)" && exit 1 ;;
+    esac
+fi
+
+# Extract architecture from platform
+case "$PLATFORM" in
+    "linux/amd64") ARCH="amd64" ;;
+    "linux/arm64") ARCH="arm64" ;;
+    *) echo "Unsupported platform: ${PLATFORM}" && exit 1 ;;
+esac
+
+# Create platform-specific image
 docker buildx build \
     --file Containerfile \
+    --platform "$PLATFORM" \
     --build-arg "COMMIT_ID=${COMMIT_ID}" \
     --build-arg "GO_VERSION=${GO_VERSION}" \
     --build-arg "VERSION=${VERSION}" \
+    --tag "boulder:${VERSION}-${ARCH}" \
     --tag "boulder:${VERSION}" \
     --tag "boulder:${COMMIT_ID}" \
     --tag boulder \
```

Comment:
I didn't give a ton of thought to these tags when we first wrote container-build.sh. Looking at them now it seems that there will be collisions if we start uploading multiple architectures from this script. For instance, for any given version `boulder:${VERSION}` could be one arch or another.

Probably a single `--tag "boulder:${VERSION}-${ARCH}"` would suffice and keep things simple. What do you think @jprenken ?

@sheurich I see that you repeat the string `"boulder:${VERSION}-${ARCH}"` here and on the two `docker run` commands below. Let's put that into a single `TAG` env var and use that in each location for consistency.

Reply:
Updated this. A follow-on step involving `docker buildx imagetools create -t letsencrypt/boulder:${VERSION} letsencrypt/boulder:${VERSION}-amd64 letsencrypt/boulder:${VERSION}-arm64` will be needed to glue the images into a single multi-arch.